src/engine/GroupByImpl.cpp

```diff
@@ -76,6 +79,39 @@ class LazyGroupByRange : public ad_utility::InputRangeMixin<
     AD_CONTRACT_CHECK(parent_ != nullptr);
   }
 
+  void initialise() {
+    lazyGroupBy_ = std::make_unique<LazyGroupBy>(
+        currentLocalVocab_, std::move(aggregateAliases_),
+        parent_->getExecutionContext()->getAllocator(), groupByCols_.size());
+    range_ =
+        Result::LazyResult(ad_utility::CachingContinuableTransformInputRange(
+            subresult_->idTables(),
+            [this](IdTableVocabPair& pair) { return process(pair); }));
+    rangeIt_ = ql::ranges::begin(range_.value());
+  }
+
+  std::optional<IdTableVocabPair> get() {
+    if (isFinished_) {
+      return std::nullopt;
+    }
+
+    if (!range_.has_value()) {
+      initialise();
+    } else {
+      ++(rangeIt_.value());
+    }
+
+    // if the main range is empty, yield the final value.
+    if (rangeIt_.value() == ql::ranges::end(range_.value())) {
+      isFinished_ = true;
+      return yieldFinalValue();
+    }
+
+    return std::make_optional<IdTableVocabPair>(std::move(*(rangeIt_.value())));
```

Comment:
```suggestion
    return std::optional{std::move(*(rangeIt_.value()))};
```

I believe you should be able to just use the default constructor here.